src/Database/Type/DecimalType.php

```diff
@@ -100,10 +101,34 @@ public function toDatabase($value, Driver $driver)
     public function toPHP($value, Driver $driver)
     {
         if ($value === null) {
-            return null;
+            return $value;
+        }
+
+        // Using coercion is faster than casting
+        // @codingStandardsIgnoreStart
+        return (float)+$value;
+        // @codingStandardsIgnoreEnd
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return array
+     */
+    public function manyToPHP(array $values, array $fields, Driver $driver)
+    {
+        foreach ($fields as $field) {
+            if (!isset($values[$field])) {
+                continue;
+            }
+
+            // Using coercion is faster than casting
+            // @codingStandardsIgnoreStart
+            $values[$field] = (float)+$values[$field];
```

Comment:
Still remains.

Reply:
Forgot that this was here, only removed in FloatType